### PR TITLE
Analyzer annotation interface support

### LINKED ISSUE #131: JSON schema validation error when including annotations for @interface.
The Analyzer does not seem to work well with annotation interfaces. Results in the output JSON to contain a type without a FullName.
Resulting error:
```
net.jimblackler.jsonschemafriend.ListValidationException: Validation errors: {"Type":"java.lang.annotation.Retention","Name":"Retentio... at #/3 failed against file:/C:/.../ISEP-LivingDocumentation/analyzer-java/target/classes/jsonschema/schema.json#/items with "Missing property FullName"
"java.lang.annotation.Retention" at #/3/Type failed against file:/C:/.../ISEP-LivingDocumentation/analyzer-java/target/classes/jsonschema/schema.json#/items/properties/Type with "Expected: [integer] Found: [string]"
[{"Name":"value","Type":"java.lang.annotation.RetentionPo... at #/3/Arguments failed against file:/C:/.../ISEP-LivingDocumentation/analyzer-java/target/classes/jsonschema/schema.json#/items/additionalProperties with "False"
"Retention" at #/3/Name failed against file:/C:/.../ISEP-LivingDocumentation/analyzer-java/target/classes/jsonschema/schema.json#/items/additionalProperties with "False"
	at net.jimblackler.jsonschemafriend.Validator.validate(Validator.java:715)
	at com.infosupport.ldoc.analyzerj.Analyzer.validateJson(Analyzer.java:59)
	at com.infosupport.ldoc.analyzerj.Analyzer.analyze(Analyzer.java:90)
	at com.infosupport.ldoc.analyzerj.Main.main(Main.java:42)
```

Reproducing error:
annotation interface with added annotation, e.g.,:
```
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface AnnotationTest {

  String testField() default "";
}
```

### PR DESCRIPTION
**Pull request by GuusVink**
Fixes #131 

Maps @interface (AnnotationDeclaration) to a plain Interface. The members of @interface (AnnotationMemberDeclaration) to Fields where the default value is represented by the InitialValue/Initializer.

For both mappings included an ADD/ADR. Note that these ADDs and their argumentation are non-trivial.

### CODE REVIEW

File: analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java [hunk 1]
```diff
@@ -146,6 +149,36 @@ public List<Description> visit(EnumDeclaration n, Analyzer arg) {
...
+   */
+  @Override
+  public List<Description> visit(AnnotationMemberDeclaration n, Analyzer arg) {
+    return List.of(
+        new FieldDescription(
+            new MemberDescription(n.getNameAsString(), Modifier.NONE.mask(),
+                visit(n.getAnnotations(), arg), n.getComment()
+                    .flatMap(c -> c.accept(this, arg).stream().findFirst()).orElse(null)),
+                resolve(n.getType()),
+                n.getDefaultValue().map(Objects::toString).orElse(null)
```
Comment (wandernauta) [inline analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java:177]: Nit: If you write this as `Object::toString`, you don't need the extra import.

**Review by wandernauta — COMMENTED** [review]
Great! Well done on the test coverage and updating the ADRs and JSON docs as well. I've put three small suggestions.

**Comment by GuusVink** [comment]
Thanks for the feedback! Good catches, I have addressed all points.

**Review by wandernauta — APPROVED** [review]
Looks good! Approved.

**Pull request merged**